core/src/core/pmmr.rs

```diff
@@ -421,14 +426,17 @@ where
 		// loop going up the tree, from node to parent, as long as we stay inside
 		// the tree.
 		let mut to_prune = vec![];
+
 		let mut current = position;
-		while current + 1 < self.last_pos {
+		while current + 1 <= self.last_pos {
 			let (parent, sibling, _) = family(current);
+
+			to_prune.push(current);
```

Comment:
Logic changed because we "remove" roots, but the hash is still available in the underlying file.
Push `current` to the prune list _before_ we check for the parent in the tree.